wtxmgr/query.go

```diff
@@ -0,0 +1,462 @@
+			str := "saved credit index exceeds number of outputs"
+			return nil, storeError(ErrData, str, nil)
+		}
+
+		// Set the Spent field since this is not done by the iterator.
+		it.elem.Spent = existsRawUnminedInput(ns, it.ck) != nil
+		details.Credits = append(details.Credits, it.elem)
+	}
+	if it.err != nil {
+		return nil, it.err
+	}
+
+	// Debit records are not saved for unmined transactions.  Instead, they
+	// must be looked up for each transaction input manually.  There are two
+	// kinds of previous credits that may be debited by an unmined
+	// transaction: mined unspent outputs (which remain marked unspent even
+	// when spent by an unmined transaction), and credits from other unmined
+	// transactions.  Both situations must be considered.
+	for i, output := range details.MsgTx.TxIn {
+		opKey := canonicalOutPoint(&output.PreviousOutPoint.Hash,
+			output.PreviousOutPoint.Index)
+		credKey := existsRawUnspent(ns, opKey)
+		if credKey != nil {
+			v := existsRawCredit(ns, credKey)
+			amount, err := fetchRawCreditAmount(v)
+			if err != nil {
+				return nil, err
+			}
+
+			details.Debits = append(details.Debits, DebitRecord{
+				Amount: amount,
+				Index:  uint32(i),
+			})
+			continue
+		}
+
+		v := existsRawUnminedCredit(ns, opKey)
+		if v == nil {
+			continue
+		}
+
+		amount, err := fetchRawCreditAmount(v)
+		if err != nil {
+			return nil, err
+		}
+		details.Debits = append(details.Debits, DebitRecord{
+			Amount: amount,
+			Index:  uint32(i),
+		})
+	}
+
+	return &details, nil
+}
+
+// RangeTransactions runs the function f on all transaction details between
+// blocks on the best chain over the height range [begin,end].  The special
+// height -1 may be used to also include unmined transactions.  If the end
+// height comes before the begin height, blocks are iterated in reverse order
+// and unmined transactions (if any) are processed first.
+//
+// The function f may return an error which, if non-nil, is propagated to the
+// caller.  Additionally, a boolean return value allows exiting the function
+// early without reading any additional transactions early when true.
+//
+// All calls to f are guaranteed to be passed a slice with more than zero
+// elements.  The slice may be reused for multiple blocks, so it is not safe to
+// use it after the loop iteration it was acquired.
+func (s *Store) RangeTransactions(begin, end int32, f func([]TxDetails) (bool, error)) error {
+	return scopedView(s.namespace, func(ns walletdb.Bucket) error {
+		var addedUnmined bool
+		if begin < 0 {
+			brk, err := s.rangeUnminedTransactions(ns, f)
+			if err != nil || brk {
+				return err
+			}
+			addedUnmined = true
+		}
+
+		brk, err := s.rangeBlockTransactions(ns, begin, end, f)
+		if err == nil && !brk && !addedUnmined && end < 0 {
+			_, err = s.rangeUnminedTransactions(ns, f)
+		}
+		return err
+	})
+}
+
+func (s *Store) rangeUnminedTransactions(ns walletdb.Bucket, f func([]TxDetails) (bool, error)) (brk bool, err error) {
```

Comment:
Thanks.

For what it's worth, I had no idea that brk was intended to mean break when reading it until I read the implementation.  Had I been looking at that in godoc (which I often do locally with the internal flag enabled so I can see all unexported functions), I would not have known what it was for since there is no comment.

As you said, no reason to debate it, but I did want to point out that it did not have the desired effect.